### PR TITLE
Port TG weapon animations

### PR DESCRIPTION
**Pull request by metalgearsloth**
Although the component doesn't seem to use the state delays so if someone knows what I need to do to resolve it that'd be great...

### CODE REVIEW

File: Resources/Textures/Effects/weapons/arcs.rsi/meta.json [hunk 1]
```diff
@@ -4,7 +4,8 @@
         "x": 32,
         "y": 32
     },
-
+    "license": "CC-BY-SA-3.0 for bite->smash",
```
Comment (ComicIronic) [inline Resources/Textures/Effects/weapons/arcs.rsi/meta.json:7]: This is a bit suspicious - the order of states is not guaranteed.

Comment (RemieRichards) [reply Resources/Textures/Effects/weapons/arcs.rsi/meta.json:7]: time to split the files
tg_arcs.rsi

Comment (ComicIronic) [reply Resources/Textures/Effects/weapons/arcs.rsi/meta.json:7]: I would support that - separates the copyright info nicely.

Comment (PJB3005) [reply Resources/Textures/Effects/weapons/arcs.rsi/meta.json:7]: FYI, `license` *should* be an SPDX license identifier so custom wording like this is a no-no.

File: Resources/Prototypes/Entities/Effects/weapon_arc.yml [hunk 2]
```diff
@@ -9,3 +9,15 @@
...
+  id: WeaponTGArc
+  save: false
+  abstract: true
+  components:
+  - type: Sprite
+    sprite: Effects/weapons/tg_arcs.rsi
+    directional: false
+    offset: 0.85, 0
+    drawdepth: Overlays
+  - type: MeleeWeaponArcAnimation
```
Comment (RemieRichards) [inline Resources/Prototypes/Entities/Effects/weapon_arc.yml:23]:
```suggestion
- type: entity
  id: WeaponTGArc
  parent: WeaponArc
  components:
  - type: Sprite
    sprite: Effects/weapons/tg_arcs.rsi
```

**Pull request merged**